### DISCUSSION TITLE
Make partition numbers / IDs stable in fdisk

### DISCUSSION THREAD

**Discussion by Strykar** [util-linux/util-linux#2670] [Ideas]
Could we make fdisk's internal identifiers stable since that is how they are called via the interactive menu?
New FS can be given sequential or any other ID, why does it need to change?
Ex: `142 Linux Extended Boot` vs `ea00` which doesn't change in gdisk.

**Comment by t-8ch** [comment]
This question is not clear.

Are you talking about partition numbers, partition identification in general, partition types or filesystem label/UUID?

All of them are mentioned in the question.

**Comment by Strykar** [comment]
I apologize, not FS, I actually meant to say partition type.
Specifically this came about after reading it is unstable in fdisk, but stable in gdisk - https://wiki.archlinux.org/title/Systemd-boot#Installation_using_XBOOTLDR

**Reply by t-8ch** [reply]
Where does it say it's unstable for fdisk?

You can use the full UUID or the name (`Linux extended boot`) in fdisk.

**Reply by Strykar** [reply]
The number ascribed to it _is_ apparently, `142`, as in the case of `Linux extended boot`.

Why is the `142` unstable?
It's a lot easier to recall a 3 or 4 digit number than a GUID.

**Reply by t-8ch** [reply]
The number is just an dynamic enumeration.

For often-used types it would be possible to add a shortcut like `xbootldr`

**Reply by Strykar** [reply]
The number is easier to recall then trying to remember _is it_ `linux extended boot`, _in title case, or just capital_ `L`.
Yes please, a shortcut like `xbootldr` would be awesome!

**Reply by t-8ch** [reply]
`L` is already generic linux partition.
I used `X`.
